### PR TITLE
msccl: enable profiling for msccl

### PR DESCRIPTION
**Pull request by nusislam**

### CODE REVIEW

File: src/collectives/device/msccl_kernel_impl.h [hunk 1]
```diff
@@ -298,9 +305,13 @@ __device__ __forceinline__ void mscclRunInterpreter(
         int thisCount = min(maxAllowedCount, count - c);
         int thisNelem = nelem * thisCount;
         if (t->type == MSCCL_SEND)
```
Comment (wenkaidu) [inline src/collectives/device/msccl_kernel_impl.h:307]: need { ... }

Comment (nusislam) [reply src/collectives/device/msccl_kernel_impl.h]: My bad..had it in my local version. Just changed.

File: src/init.cc [hunk 2]
```diff
@@ -353,7 +353,11 @@ static ncclResult_t commFree(ncclComm_t comm) {
   struct ncclProf *prof, *prof_seq;
   prof = (struct ncclProf*)malloc(sizeof(struct ncclProf)*MAXCHANNELS*PROFILE_NUM_LAUNCHES);
   CUDACHECK(hipMemcpy(prof, comm->devComm->devProf, sizeof(struct ncclProf)*MAXCHANNELS*PROFILE_NUM_LAUNCHES, hipMemcpyDeviceToHost));
-  #define VEGA_GPU_RTC_FREQUENCY 2.5E7
+#if defined(__gfx940__)
```
Comment (wenkaidu) [inline src/init.cc:356]: compilation time detection doesn't work on host side. need to query device property

**Comment by nusislam** [comment]
This can be closed as a different version of the change has been merged to RCCL.

**Pull request closed without merging**